In [23]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [24]:
df = pd.read_csv("data/Kickstarter000.csv")

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3779 entries, 0 to 3778
Data columns (total 37 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   backers_count             3779 non-null   int64  
 1   blurb                     3779 non-null   object 
 2   category                  3779 non-null   object 
 3   converted_pledged_amount  3779 non-null   int64  
 4   country                   3779 non-null   object 
 5   created_at                3779 non-null   int64  
 6   creator                   3779 non-null   object 
 7   currency                  3779 non-null   object 
 8   currency_symbol           3779 non-null   object 
 9   currency_trailing_code    3779 non-null   bool   
 10  current_currency          3779 non-null   object 
 11  deadline                  3779 non-null   int64  
 12  disable_communication     3779 non-null   bool   
 13  friends                   8 non-null      object 
 14  fx_rate 

In [26]:
df.iloc[:, 3:16].head()

converted_pledged_amount country  created_at  \
0                       802      US  1387659690   
1                      2259      US  1549659768   
2                     29638      US  1477242384   
3                     49158      IT  1540369920   
4                       549      US  1425706517   

                                             creator currency currency_symbol  \
0  {"id":1495925645,"name":"Daniel","is_registere...      USD               $   
1  {"id":1175589980,"name":"Katherine","slug":"fr...      USD               $   
2  {"id":1196856269,"name":"MelissaThomas","is_re...      USD               $   
3  {"id":1569700626,"name":"WAO","slug":"wearewao...      EUR               €   
4  {"id":1870845385,"name":"Kalpit Jain","is_regi...      USD               $   

   currency_trailing_code current_currency    deadline  disable_communication  \
0                    True              USD  1391899046                  False   
1                    True              USD  1551801611                  False   
2                    True              USD  1480607930                  False   
3                   False              USD  1544309940                  False   
4                    True              USD  1428511017                  False   

  friends   fx_rate     goal  
0     NaN  1.000000    200.0  
1     NaN  1.000000    400.0  
2     NaN  1.000000  27224.0  
3     NaN  1.128433  40000.0  
4     NaN  1.000000   1000.0

 0   backers_count             number of backers 
 1   blurb                     short description 
 2   category                  Kickstarter categories 
 3   converted_pledged_amount  pledged amount in USD  
 4   country                   country 
 5   created_at                creation date/time?  
 6   creator                   Kickstarter account that created the project 
 7   currency                  currency the project asks for
 8   currency_symbol           currency symbol 
 9   currency_trailing_code    ?   
 10  current_currency          ?
 11  deadline                  deadline date/time?
 12  disable_communication     don't allow communication/comments   
 13  friends                   drop (not enough data) 
 14  fx_rate                   currency conversion rate
 15  goal                      fixed amount required for funding
 16  id                        Kickstarter project id  
 17  is_backing                drop (not enough data)  
 18  is_starrable              ?  
 19  is_starred                drop (not enough data)  
 20  launched_at               launch date/time?  
 21  location                  location
 22  name                      project name 
 23  permissions               drop (not enough data) 
 24  photo                     maybe drop?
 25  pledged                   pledged amount in original currency
 26  profile                   drop? 
 27  slug                      drop (converted project name) 
 28  source_url                source url - drop 
 29  spotlight                 drop? (probably not influencable by the creators)   
 30  staff_pick                drop? (probably not influencable by the creators)   
 31  state                     !!!!!target!!!!! 
 32  state_changed_at          state changed at date/time?  
 33  static_usd_rate           conversion rate -> drop
 34  urls                      drop 
 35  usd_pledged               drop
 36  usd_type                  drop

In [27]:
df['current_currency'].unique()

array(['USD', 'CAD', 'GBP', 'AUD'], dtype=object)

In [28]:
no_comm = df[df['disable_communication'] == True]

In [29]:
len(no_comm)

10

In [30]:
comm = df[df['disable_communication'] == False]

In [31]:
len(comm)

3769

### location example

{"id":2379574,
"name":"Chicago",
"slug":"chicago-il",
"short_name":"Chicago, IL",
"displayable_name":"Chicago, IL",
"localized_name":"Chicago",
"country":"US",
"state":"IL",
"type":"Town",
"is_root":false,
"urls":{"web":{"discover":"https://www.kickstarter.com/discover/places/chicago-il","location":"https://www.kickstarter.com/locations/chicago-il"},"api":{"nearby_projects":"https://api.kickstarter.com/v1/discover?signature=1552595044.c1041c6bca69b0b72738f3b9504ebf921b3e5e0e&woe_id=2379574"}}}'

In [32]:
df['category'].unique()

array(['{"id":43,"name":"Rock","slug":"music/rock","position":17,"parent_id":14,"color":10878931,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/music/rock"}}}',
       '{"id":54,"name":"Mixed Media","slug":"art/mixed media","position":6,"parent_id":1,"color":16760235,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/art/mixed%20media"}}}',
       '{"id":280,"name":"Photobooks","slug":"photography/photobooks","position":5,"parent_id":15,"color":58341,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/photography/photobooks"}}}',
       '{"id":266,"name":"Footwear","slug":"fashion/footwear","position":5,"parent_id":9,"color":16752598,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/fashion/footwear"}}}',
       '{"id":51,"name":"Software","slug":"technology/software","position":11,"parent_id":16,"color":6526716,"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/techno

### category example

{"id":43,
"name":"Rock",
"slug":"music/rock",
"position":17,
"parent_id":14,
"color":10878931,
"urls":{"web":{"discover":"http://www.kickstarter.com/discover/categories/music/rock"}}}'

In [33]:
df['creator'].unique()

array(['{"id":1495925645,"name":"Daniel","is_registered":null,"chosen_currency":null,"avatar":{"thumb":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=40&h=40&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3d655afafac9dbb59c1e675adfa87082","small":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=160&h=160&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3973d24f5c3db1ed1d5c84cec8af1d6d","medium":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=160&h=160&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3973d24f5c3db1ed1d5c84cec8af1d6d"},"urls":{"web":{"user":"https://www.kickstarter.com/profile/1495925645"},"api":{"user":"https://api.kickstarter.com/v1/users/1495925645?signature=1552621462.382152be8688b069cea0ab5f3a266a1530d83c3f"}}}',
       '{"id":1175589980,"name":"Katherine","slug":"freckleddisney","is_

### creator example

{"id":1495925645,
"name":"Daniel",
"is_registered":null,
"chosen_currency":null,
"avatar":{"thumb":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=40&h=40&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3d655afafac9dbb59c1e675adfa87082","small":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=160&h=160&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3973d24f5c3db1ed1d5c84cec8af1d6d","medium":"https://ksr-ugc.imgix.net/assets/006/041/047/c44d1a95c2139ae46af635c7c6e7ea76_original.jpg?ixlib=rb-1.1.0&w=160&h=160&fit=crop&v=1461362658&auto=format&frame=1&q=92&s=3973d24f5c3db1ed1d5c84cec8af1d6d"},"urls":{"web":{"user":"https://www.kickstarter.com/profile/1495925645"},"api":{"user":"https://api.kickstarter.com/v1/users/1495925645?signature=1552621462.382152be8688b069cea0ab5f3a266a1530d83c3f"}}}',

In [34]:
df['photo'].isna().sum()

0

In [35]:
df['profile'].unique()

array(['{"id":822687,"project_id":822687,"state":"inactive","state_changed_at":1425915845,"name":null,"blurb":null,"background_color":null,"text_color":null,"link_background_color":null,"link_text_color":null,"link_text":null,"link_url":null,"show_feature_image":false,"background_image_opacity":0.8,"should_show_feature_image_section":true,"feature_image_attributes":{"image_urls":{"default":"https://ksr-ugc.imgix.net/assets/011/625/534/5bea1760d7f20943c4cd5e9b4911c1bd_original.jpg?ixlib=rb-1.1.0&crop=faces&w=1552&h=873&fit=crop&v=1463685705&auto=format&frame=1&q=92&s=90c72b785ef97539099dffb8531dcad3","baseball_card":"https://ksr-ugc.imgix.net/assets/011/625/534/5bea1760d7f20943c4cd5e9b4911c1bd_original.jpg?ixlib=rb-1.1.0&crop=faces&w=560&h=315&fit=crop&v=1463685705&auto=format&frame=1&q=92&s=883b3ff098e5fc4cf6fec280665f5fd0"}}}',
       '{"id":3665299,"project_id":3665299,"state":"inactive","state_changed_at":1549659768,"name":null,"blurb":null,"background_color":null,"text_color":null,

### profile example

{"id":822687,
"project_id":822687,
"state":"inactive",
"state_changed_at":1425915845,
"name":null,
"blurb":null,
"background_color":null,
"text_color":null,
"link_background_color":null,
"link_text_color":null,
"link_text":null,
"link_url":null,
"show_feature_image":false,
"background_image_opacity":0.8,
"should_show_feature_image_section":true,
"feature_image_attributes":{"image_urls":{"default":"https://ksr-ugc.imgix.net/assets/011/625/534/5bea1760d7f20943c4cd5e9b4911c1bd_original.jpg?ixlib=rb-1.1.0&crop=faces&w=1552&h=873&fit=crop&v=1463685705&auto=format&frame=1&q=92&s=90c72b785ef97539099dffb8531dcad3","baseball_card":"https://ksr-ugc.imgix.net/assets/011/625/534/5bea1760d7f20943c4cd5e9b4911c1bd_original.jpg?ixlib=rb-1.1.0&crop=faces&w=560&h=315&fit=crop&v=1463685705&auto=format&frame=1&q=92&s=883b3ff098e5fc4cf6fec280665f5fd0"}}}'

## list of features we keep

 1   blurb                     short description 
 2   category                  Kickstarter categories  
 4   country                   country 
 5   created_at                creation date/time?  
 7   currency                  currency the project asks for
 10  current_currency          ?
 11  deadline                  deadline date/time?
 12  disable_communication     don't allow communication/comments  
|14  fx_rate                   currency conversion rate 
 15  goal                      fixed amount required for funding (convert with fx_rate)
 18  is_starrable              ?  
 20  launched_at               launch date/time?  
 21  location                  location
 22  name                      project name 
 24  photo                     -> check if projects without photo
 31  state                     !!!!!target!!!!! 

## stakeholder

* who? - people/creators who are considering launching a project on Kickstarter
* why? - to find out if it's worth investing the time/money in creating materials/launching a project and which criteria to consider in order to make it successful
* metric? - f_beta (probably imbalanced data)
* model: classifier (binary)


# EDA - Round 1

In [36]:
# load data from the first 5 csv-files and concatenate

df1 = pd.read_csv('data/Kickstarter000.csv')
df2 = pd.read_csv('data/Kickstarter001.csv')
df3 = pd.read_csv('data/Kickstarter002.csv')
df4 = pd.read_csv('data/Kickstarter003.csv')
df5 = pd.read_csv('data/Kickstarter004.csv')

# concatenate

df = pd.concat([df1, df2, df3, df4, df5], axis=0, ignore_index=True)

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18937 entries, 0 to 18936
Data columns (total 37 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   backers_count             18937 non-null  int64  
 1   blurb                     18936 non-null  object 
 2   category                  18937 non-null  object 
 3   converted_pledged_amount  18937 non-null  int64  
 4   country                   18937 non-null  object 
 5   created_at                18937 non-null  int64  
 6   creator                   18937 non-null  object 
 7   currency                  18937 non-null  object 
 8   currency_symbol           18937 non-null  object 
 9   currency_trailing_code    18937 non-null  bool   
 10  current_currency          18937 non-null  object 
 11  deadline                  18937 non-null  int64  
 12  disable_communication     18937 non-null  bool   
 13  friends                   26 non-null     object 
 14  fx_rat

In [37]:
df.head()

backers_count                                              blurb  \
0             21  2006 was almost 7 years ago.... Can you believ...   
1             97  An adorable fantasy enamel pin series of princ...   
2             88  Helping a community come together to set the s...   
3            193  Every revolution starts from the bottom and we...   
4             20  Learn to build 10+ Applications in this comple...   

                                            category  \
0  {"id":43,"name":"Rock","slug":"music/rock","po...   
1  {"id":54,"name":"Mixed Media","slug":"art/mixe...   
2  {"id":280,"name":"Photobooks","slug":"photogra...   
3  {"id":266,"name":"Footwear","slug":"fashion/fo...   
4  {"id":51,"name":"Software","slug":"technology/...   

   converted_pledged_amount country  created_at  \
0                       802      US  1387659690   
1                      2259      US  1549659768   
2                     29638      US  1477242384   
3                     49158      IT  1540369920   
4                       549      US  1425706517   

                                             creator currency currency_symbol  \
0  {"id":1495925645,"name":"Daniel","is_registere...      USD               $   
1  {"id":1175589980,"name":"Katherine","slug":"fr...      USD               $   
2  {"id":1196856269,"name":"MelissaThomas","is_re...      USD               $   
3  {"id":1569700626,"name":"WAO","slug":"wearewao...      EUR               €   
4  {"id":1870845385,"name":"Kalpit Jain","is_regi...      USD               $   

   currency_trailing_code  ...  \
0                    True  ...   
1                    True  ...   
2                    True  ...   
3                   False  ...   
4                    True  ...   

                                                slug  \
0                              new-final-round-album   
1                    princess-pals-enamel-pin-series   
2  their-life-through-their-lens-the-amish-and-me...   
3                           wao-the-eco-effect-shoes   
4                     apple-watch-development-course   

                                          source_url  spotlight staff_pick  \
0  https://www.kickstarter.com/discover/categorie...       True      False   
1  https://www.kickstarter.com/discover/categorie...       True      False   
2  https://www.kickstarter.com/discover/categorie...       True       True   
3  https://www.kickstarter.com/discover/categorie...       True      False   
4  https://www.kickstarter.com/discover/categorie...      False      False   

        state  state_changed_at  static_usd_rate  \
0  successful        1391899046         1.000000   
1  successful        1551801611         1.000000   
2  successful        1480607932         1.000000   
3  successful        1544309940         1.136525   
4      failed        1428511019         1.000000   

                                                urls  usd_pledged  \
0  {"web":{"project":"https://www.kickstarter.com...    802.00000   
1  {"web":{"project":"https://www.kickstarter.com...   2259.00000   
2  {"web":{"project":"https://www.kickstarter.com...  29638.00000   
3  {"web":{"project":"https://www.kickstarter.com...  49075.15252   
4  {"web":{"project":"https://www.kickstarter.com...    549.00000   

        usd_type  
0  international  
1  international  
2  international  
3  international  
4       domestic  

[5 rows x 37 columns]

In [38]:
df.columns

Index(['backers_count', 'blurb', 'category', 'converted_pledged_amount',
       'country', 'created_at', 'creator', 'currency', 'currency_symbol',
       'currency_trailing_code', 'current_currency', 'deadline',
       'disable_communication', 'friends', 'fx_rate', 'goal', 'id',
       'is_backing', 'is_starrable', 'is_starred', 'launched_at', 'location',
       'name', 'permissions', 'photo', 'pledged', 'profile', 'slug',
       'source_url', 'spotlight', 'staff_pick', 'state', 'state_changed_at',
       'static_usd_rate', 'urls', 'usd_pledged', 'usd_type'],
      dtype='object')

In [39]:
# drop columns we won't be using

df = df[['backers_count', 'blurb', 'category', 'converted_pledged_amount',
       'country', 'created_at', 'currency', 'deadline',
       'disable_communication', 'fx_rate', 'goal', 'is_starrable', 'launched_at', 'location',
       'name', 'photo', 'spotlight', 'staff_pick', 'state']]

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18937 entries, 0 to 18936
Data columns (total 19 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   backers_count             18937 non-null  int64  
 1   blurb                     18936 non-null  object 
 2   category                  18937 non-null  object 
 3   converted_pledged_amount  18937 non-null  int64  
 4   country                   18937 non-null  object 
 5   created_at                18937 non-null  int64  
 6   currency                  18937 non-null  object 
 7   deadline                  18937 non-null  int64  
 8   disable_communication     18937 non-null  bool   
 9   fx_rate                   18937 non-null  float64
 10  goal                      18937 non-null  float64
 11  is_starrable              18937 non-null  bool   
 12  launched_at               18937 non-null  int64  
 13  location                  18917 non-null  object 
 14  name  

### Missing Values

* location has 20 missing values (out of 18937)
* blurb has 1 missing value (out of 18937)

In [40]:
#sns.pairplot(df)

In [41]:
df['state'].unique()

array(['successful', 'failed', 'live', 'canceled', 'suspended'],
      dtype=object)

In [42]:
live = df.query('state == "live"')
len(live)

643

In [43]:
canceled = df.query('state == "canceled"')
len(canceled)

783

In [44]:
suspended = df.query('state == "suspended"')
len(suspended)

58

In [45]:
successful = df.query('state == "successful"')
len(successful)

10379

In [46]:
failed = df.query('state == "failed"')
len(failed)

7074

In [47]:
# convert 'state' to numerical
# successful: 1
# failed: 0
# drop: live, suspended, canceled

df = df.query('state != "live"')
df = df.query('state != "suspended"')
df = df.query('state != "canceled"')
print(df['state'].unique()) # check that 'state' only contains failed and successful

df['state'].replace({'failed':0, 'successful':1}, inplace=True)
print(df['state'].unique()) # check that 'state' only contains 1 and 0


['successful' 'failed']
[1 0]


In [ ]:
# plot frequency of success and failure

sns.countplot()


